annotations/openshift-annotations/src/main/java/io/ap4k/openshift/hook/OcBuildHook.java

```diff
@@ -17,29 +17,53 @@
 package io.ap4k.openshift.hook;
 
 import io.ap4k.Ap4kException;
+import io.ap4k.deps.kubernetes.api.model.HasMetadata;
+import io.ap4k.deps.kubernetes.api.model.KubernetesList;
+import io.ap4k.deps.kubernetes.api.model.Secret;
+import io.ap4k.deps.openshift.api.model.BuildConfig;
+import io.ap4k.deps.openshift.api.model.ImageStream;
+import io.ap4k.deps.openshift.client.DefaultOpenShiftClient;
+import io.ap4k.deps.openshift.client.OpenShiftClient;
 import io.ap4k.hook.ProjectHook;
+import io.ap4k.openshift.config.S2iConfig;
+import io.ap4k.openshift.util.OpenshiftUtils;
 import io.ap4k.project.Project;
+import io.ap4k.utils.Serialization;
 
 import java.io.File;
-import java.io.FileNotFoundException;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
 
 public class OcBuildHook extends ProjectHook {
```

Comment:
Wouldn't it be better to name this `JavaBuildHook`?

Reply:
It uses oc to perform the actual build.

Reply:
Yeah... stupid me :P. Looking at the `init` method threw me off, I didn't see the `run` method